src/aap_eda/api/serializers/eda_credential.py

```diff
@@ -151,6 +151,18 @@ def validate(self, data):
             id=data.get("credential_type_id")
         )
 
+        # Analytics only allows one credential
+        if (
+            credential_type.name in enums.SINGLETON_CREDENTIAL_TYPES
+            and models.EdaCredential.objects.filter(
+                credential_type=credential_type
+            ).exists()
+        ):
+            raise serializers.ValidationError(
+                "Only one credential is allowed for analytics type: "
```

Comment:
`SINGLETON_CREDENTIAL_TYPES` doesn't suggest analytics. So the word analytics should be removed from the message.